exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -115,6 +118,29 @@ const (
 	Stable ReleaseChannel = "stable"
 )
 
+// MasterAuthorizedNetworksConfig contains configuration options for the master authorized networks feature.
+// Enabled master authorized networks will disallow all external traffic to access
+// Kubernetes master through HTTPS except traffic from the given CIDR blocks,
+// Google Compute Engine Public IPs and Google Prod IPs.
+type MasterAuthorizedNetworksConfig struct {
+	// Whether or not master authorized networks is enabled.
+	Enabled bool `json:"enabled,omitempty"`
+	// cidr_blocks define up to 50 external networks that could access
+	// Kubernetes master through HTTPS.
+	CidrBlocks []*MasterAuthorizedNetworksConfigCidrBlock `json:"cidr_blocks,omitempty"`
+	// Whether master is accessible via Google Compute Engine Public IP addresses.
+	GcpPublicCidrsAccessEnabled *bool `json:"gcp_public_cidrs_access_enabled,omitempty"`
+}
+
+// MasterAuthorizedNetworksConfigCidrBlock contains an optional name and one CIDR block.
+type MasterAuthorizedNetworksConfigCidrBlock struct {
+	// display_name is an optional field for users to identify CIDR blocks.
+	DisplayName string `json:"display_name,omitempty"`
```

Comment:
If this is optional could we add the `+optional` field validation comment?

Reply:
display name is not optional and I am fixing the comment.